Add `no-clown-fiesta-theme` recipe

ranmaru22 opened the pull request:
### Brief summary of what the package does

A not-so-colorful theme for Emacs. Based on the [Neovim theme of the same name](https://github.com/aktersnurra/no-clown-fiesta.nvim) by aktersnurra. Comes with out-of-the-box support for a few common packages and an optional accompanying Treemacs theme.

### Direct link to the package repository

https://github.com/ranmaru22/no-clown-fiesta-theme.el

### Your association with the package

Maintainer.

### Relevant communications with the upstream package maintainer

None needed.

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  Quick first pass  (I suspect the treemacs file is meant to be excluded in the recipe?) --

`no-clown-fiesta-theme.el` with byte-compile using Emacs 28.1:
- Temporarily ignoring `no-byte-compile` flag
```
no-clown-fiesta-theme.el:45:2: Warning: Unused lexical variable `bg-darker'
no-clown-fiesta-theme.el:45:2: Warning: Unused lexical variable `dark-purple'
no-clown-fiesta-theme.el:45:2: Warning: Unused lexical variable `info-yellow'
no-clown-fiesta-theme.el:45:2: Warning: Unused lexical variable `hint-green'
no-clown-fiesta-theme.el:45:2: Warning: Unused lexical variable `magit-blue'
```
fyi; if you plan to use the byte-compiler to help check correctness and don't want these warnings, I believe you can rename them to e.g. `_bg-darker` to hint that they're not (currently) used.

`no-clown-fiesta-theme.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- no-clown-fiesta-theme.el#L5: Prefer `;; Version` over `;; Package-Version` (MELPA automatically adds `Package-Version`)
```

`no-clown-fiesta-treemacs.el` with byte-compile using Emacs 28.1:
- Temporarily ignoring `no-byte-compile` flag
```
no-clown-fiesta-treemacs.el:10:24: Warning: `"jpg"' is a malformed function
no-clown-fiesta-treemacs.el:10:24: Warning: `"ttf"' is a malformed function
no-clown-fiesta-treemacs.el:10:24: Warning: `"mkv"' is a malformed function
no-clown-fiesta-treemacs.el:10:24: Warning: `"mp3"' is a malformed function
no-clown-fiesta-treemacs.el:10:24: Warning: `"zip"' is a malformed function
no-clown-fiesta-treemacs.el:10:24: Warning: `"gitignore"' is a malformed
    function
no-clown-fiesta-treemacs.el:10:24: Warning: `"key"' is a malformed function
no-clown-fiesta-treemacs.el:26:33: Warning: reference to free variable
    `root-open'
no-clown-fiesta-treemacs.el:65:81: Warning: error called with 0 arguments, but
    requires 1+
In end of data:
no-clown-fiesta-treemacs.el:108:4: Warning: the function `treemacs-load-theme'
    is not known to be defined.
no-clown-fiesta-treemacs.el:67:42: Warning: the function `warning' is not
    known to be defined.
no-clown-fiesta-treemacs.el:56:21: Warning: the function `license' is not
    known to be defined.
no-clown-fiesta-treemacs.el:51:21: Warning: the function `tag-leaf' is not
    known to be defined.
no-clown-fiesta-treemacs.el:46:21: Warning: the function `tag-closed' is not
    known to be defined.
no-clown-fiesta-treemacs.el:41:21: Warning: the function `tag-open' is not
    known to be defined.
no-clown-fiesta-treemacs.el:70:30: Warning: the function
    `all-the-icons-material' is not known to be defined.
no-clown-fiesta-treemacs.el:36:21: Warning: the function `dir-closed' is not
    known to be defined.
no-clown-fiesta-treemacs.el:31:21: Warning: the function `dir-open' is not
    known to be defined.
no-clown-fiesta-treemacs.el:26:21: Warning: the function `root-closed' is not
    known to be defined.
no-clown-fiesta-treemacs.el:55:30: Warning: the function
    `all-the-icons-octicon' is not known to be defined.
no-clown-fiesta-treemacs.el:21:21: Warning: the function `fallback' is not
    known to be defined.
no-clown-fiesta-treemacs.el:35:30: Warning: the function
    `all-the-icons-faicon' is not known to be defined.
no-clown-fiesta-treemacs.el:19:8: Warning: the function `treemacs-create-icon'
    is not known to be defined.
no-clown-fiesta-treemacs.el:16:4: Warning: the function
    `treemacs-create-theme' is not known to be defined.
```

`no-clown-fiesta-treemacs.el` with package-lint 20221003.1636:
```
2 issues found:
6:0: error: "no-clown-fiesta-treemacs--line-space-hook" doesn't start with package's prefix "no-clown-fiesta-theme".
10:1: warning: `with-eval-after-load' is for use in configurations, and should rarely be used in packages.
```

ranmaru22 commented:
Thanks for the feedback. I've addressed the issues and run the files through melpazoid again. It comes back clean now.

I've also removed the treemacs theme for now as there seems to be no clean way to implement this without actually putting treemacs as a dependency (which I obviously don't want to do).

riscy commented:
This looks good now -- very tidy looking theme.  Thanks again, and welcome to MELPA. :)
